doc/cprover-manual/memory-analyzer.md

```diff
@@ -0,0 +1,78 @@
+[CPROVER Manual TOC](../../)
+
+## Memory Analyzer
+
+The memory analyzer is a front-end for running and querying GDB in order to
+obtain a state of the input program. A common application would be to obtain a
+snapshot of a program under analysis at a particular state of execution. Such a
+snapshot could be useful on its own: to query about values of particular
+variables. Furthermore, since that snapshot is a state of a valid concrete
+execution, it can also be used for subsequent analyses.
+
+## Usage
+
+We assume that the user wants to inspect a binary executable compiled with
+debugging symbols and a symbol table information understandable by CBMC, e.g.
+(having `goto-gcc` on the `PATH`):
+
+```sh
+$ goto-gcc -g -std=c11 input_program.c -o input_program_exe
+```
+
+Calling `goto-gcc` instead of simply compiling with `gcc` produces an ELF binary
+with a goto section that contains the goto model (goto program + symbol table).
+
+The user can choose to either run GDB from a core-file `--core-file cf` or to a
+break-point `--breakpoint bp` (and only one of those). The tool also expects a
```

Comment:
So you're telling me I have a choice, but I don't yet know what the overall workflow is - and thus I cannot make an informed decision. It would be good to include a brief paragraph that sketches the steps involved at very high level (compile the program, run it without GDB and obtain a core and then load the core in GDB _or_ run it within GDB directly, and then ...).

Reply:
Explained and expanded.